src/response.js

```diff
@@ -23,12 +39,18 @@ export default class Response extends Body {
 	constructor(body = null, options = {}) {
 		super(body, options);
 
-		// eslint-disable-next-line no-eq-null, eqeqeq, no-negated-condition
-		const status = options.status != null ? options.status : 200;
+		// if status is not provided or is null, let's use default 200 value
+		// otherwise, lets check if the value is in correct range
+		const status =	(typeof options.status === 'undefined' || options.status === null) ? 200 :
+			validateStatusValue(options.status);
 
 		const headers = new Headers(options.headers);
 
-		if (body !== null && !headers.has('Content-Type')) {
+		if (status === 204 && typeof body !== 'undefined' && body !== null) {
```

Comment:
I think we also need to add `101` and `103`, due to [this note]( https://fetch.spec.whatwg.org/#dom-response-error:~:text=101%20and%20103%20are%20included%20in%20null%20body%20status%20due%20to%20their%20use%20elsewhere.%20They%20do%20not%20affect%20this%20step.)